src/xia2/Wrappers/XDS/XDS.py

```diff
@@ -472,7 +474,13 @@ def template_to_xds(template):
         # we don't know what is in the master file but we know at this point
         # that the word master is in there, so... otherwise can get complicated
         # side-effects when people have a folder named e.g. data_200.
-        return template.replace("master.h5", "??????.h5")
+
+        if template.endswith("master.h5"):
+            return template.replace("master.h5", "??????.h5")
+        elif template.endswith(".nxs"):
+            return template.replace(".nxs", "_??????.h5")
+        else:
+            return template
```

Comment:
Obviously `re` needs to be added also
```suggestion
      return re.sub(r"_master\.h5$|\.nxs$", "_??????.h5", template)
```

but - do we know why we apparently need to do this renaming? durin limitation?

Reply:
Yes, this looks better. I missed that ".nxs" can also appear in the template name, not just at the end. 

The given template name is copied into the XDS.INP file. I haven't looked into the Durin code, but I guess that is the reason. This part of the xia2  is written only for Eiger detectors, which produce a data file and a master file. The Durin plugin is probably written to deduce image indexing from the `_??????` part. I tried running the XDS.INP script with just the Nexus file (.nxs), but it fails. Fixing this on the Durin side would require more work, so this looked easier.